tracer/transport.go

```diff
@@ -153,7 +183,10 @@ func (t *httpTransport) SendServices(services map[string]Service) (*http.Respons
 		return t.SendServices(services)
 	}
 
-	response.Body.Close()
+	if sc := response.StatusCode; sc != 200 {
+		return response, fmt.Errorf("SendServices expected response code 200, received %v", sc)
```

Comment:
We weren't reporting non-200 errors from the agent, which I think we should be doing.

When we call `SendServices` in `flushServices` and `SendTraces` in `flushTraces`, we don't check the response status code. Does it make more sense to check for a 200 status code here, or in the caller?